VHAR-2288 Materiaalikirjastoon pohjatyötä

nizu-solita opened the pull request.

nizu-solita committed:
Lisätty tietokantamalli ja testidata urakkakohtaisille massoille

nizu-solita committed:
Lisätty päällystyskohteen massakirjastosta tietojen nouto ja tulostus päällystysilmoituksen etusivulle.

jarnovayrynen committed:
Merge branch 'develop' into VHAR-2288

jarnovayrynen commented on src/clj/harja/palvelin/palvelut/yllapitokohteet.clj:
```diff
@@ -702,6 +702,14 @@
+  (log/debug "hae urakan paallystemassat, " (pr-str urakka-id))
+  (oikeudet/vaadi-lukuoikeus oikeudet/urakat-kohdeluettelo-paallystysilmoitukset user urakka-id)
+  (let [massat (q/hae-urakan-paallystysmassat db {:urakka-id urakka-id})]
+    (log/debug "Data haettu, tässä se " (pr-str massat))
```
Nää ehkä vähän tällaisia devaushetkeen tarkoitettuja lokituksia, vois olla ennen mergeä parempi siivota pois. Samoin koko letin vois purkaa tässä, ja palauttaa vaan suoraan tuon (q/hae-urakan-paallystysmassat db {:urakka-id urakka-id}) niin saadaan napakka funktio

jarnovayrynen commented on tietokanta/src/main/resources/db/migration/V1_830__.sql:
```diff
@@ -0,0 +1,27 @@
+    urakka            INTEGER REFERENCES urakka (id),
+
+    massatyyppitunnus TEXT,
+    raekoko           INTEGER,
```
tässä taulun määrityksessä pistää silmään nyt jälkikäteen, että pitää analysoida vielä, mitkä kentistä olisi syytä määrittää NOT NULL

nizu-solita committed:
Muokattu päällystysmassojen nouto specql formaattiin

nizu-solita committed:
Lisätty urakan massan tallennuksen logiikka. Siirretty tallennus sekä haku-toiminnot päällystys-namespacen alle.

nizu-solita committed:
Merge branch 'develop' into VHAR-2288

nizu-solita committed:
Laitettu kevyt rakenne massakirjaston modalille valmiiksi

rckopteri committed:
Merge branch 'develop' into VHAR-2288

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_830__.sql

rckopteri committed:
Merge branch 'develop' into VHAR-2288

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_832__.sql

rckopteri committed:
Alustavat pot2 tietokantataulut

rckopteri committed:
Keskeneräisiä massakirjastohommia

jarnovayrynen committed:
Korjaa pot2_paallystystiedot_idx oikeaan tauluun

jarnovayrynen committed:
Tarkenna kuulamyllyluokka ja max raekoko sarakkeiden tyyppiä

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-2288

jarnovayrynen committed:
Korjaa NOT NULL oikeaan kohtaan määritystä

jarnovayrynen committed:
Bumppaa migraationumeroa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-2288

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-2288

jarnovayrynen committed:
Lisää pot2 komponentti testeihin

jarnovayrynen committed:
Päivitä päällystyksen testidatoja

Mm. jatka Muhoksen urakkaa jotta on yhä käynnissä

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-2288

jarnovayrynen committed:
Päivitä tietomallia, selvennä mat.kirj. ja POT2 roolijakoa

jarnovayrynen committed:
Indeksiarvo päivittyi, korjaa testi

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-2288

jarnovayrynen committed:
Päivitä runkoaineen mallia

jarnovayrynen committed:
Poista kuollutta koodia

jarnovayrynen committed:
Vie materiaalikirjasto modaalin sisään

jarnovayrynen committed:
Korjaile kenttien nimiä kysylyihin

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-2288

jarnovayrynen committed:
Näytä runkoaineet massakirjastossa näyttämiseen

jarnovayrynen committed:
Tyylittele massataulukko

jarnovayrynen committed:
Tyylittele gridiä, lisää React-key:t

jarnovayrynen committed:
Lisää pot2_massatyyppi kantaan

Tee Muokkaa urakan materiaaleja nappi speksin mukaiseksi

jarnovayrynen committed:
Tarkenna litteyslukuluokan tyyppiä kannassa

jarnovayrynen committed:
Lisää massan nimelle rivi

jarnovayrynen committed:
Uudelleennimeä kantava kerros --> alusta

jarnovayrynen committed:
Korjaa massalomakkeen parametrien paikat

jarnovayrynen committed:
Lisää komponentti ainevalinta-kentat

jarnovayrynen committed:
Suurenna checkboxin klikattavaa aluetta jos optio annettu

jarnovayrynen committed:
Lisää kenttiä runkoainevalintoihin

jarnovayrynen committed:
Tee otsikko-ja-kentta komponentti

Hioa eri ainetyyppien kenttäkokoonpanot

jarnovayrynen committed:
Tee sideainetyypin komponentti

jarnovayrynen committed:
Tuo sideainetyypit valintalistaan

jarnovayrynen committed:
Lisää tapa validoida numerokenttä, joka ei ole lomakkeessa

jarnovayrynen committed:
Näytä puuttuvat pakolliset kentät, disabloi tallenna jos lomake ei validi

jarnovayrynen committed:
Tallenna massan runkoaineet

jarnovayrynen committed:
Kierrä massan eri aineiden JOIN:ssa ilmennyt bugi

Useampi kuin yksi joini --> kaikki runkoaineet eivät näkyneet. Nyt siis
haetaan erikseen side- ja lisäaineet ja assocataan massan tietoihin.

jarnovayrynen committed:
Näytä muun runkoaineen kuvaus

jarnovayrynen committed:
Korjaa typo

jarnovayrynen committed:
Muokkaa massaa, toteuta eri ainetyyppien upsert

jarnovayrynen committed:
Rukkaa testejä

jarnovayrynen committed:
Tyylittele gridiä

jarnovayrynen committed:
Täsmennä massojen numerotyypit, tee validoinnit

jarnovayrynen committed:
Tää naiivi testi hae-urakan-pot2-massat

jarnovayrynen committed:
Toteuta sideaineen poistaminen kannasta

jarnovayrynen committed:
Validoi runko- ja sideaineet

jarnovayrynen committed:
Validoi lisäaineet

jarnovayrynen committed:
Älä tarjoa jo valittuja sideainetyyppejä uudestaan

jarnovayrynen committed:
Lisää kyky poistaa massa ja varmistusdialogi sille

jarnovayrynen committed:
VHAR-2743 Duplikoi/Kopioi massa

jarnovayrynen committed:
Bumppaa migraationumeroa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-2288

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-2288

jarnovayrynen committed:
VHAR-3183 Paranna massakirjaston visuaalista ulkoasua

jarnovayrynen committed:
Pompauta migraationumeroa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-2288

 Conflicts:
	src/cljs/harja/tiedot/urakka/urakka.cljs
	src/cljs/harja/ui/kentat.cljs
	src/cljs/harja/ui/lomake.cljs

jarnovayrynen committed:
Piilota materiaalikirjasto väliaikaisesti developiin mergen vuoksi

Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-2288

# Conflicts:
#	src/cljs/harja/tiedot/urakka/urakka.cljs
#	src/cljs/harja/ui/kentat.cljs
#	src/cljs/harja/ui/lomake.cljs

jarnovayrynen committed:
Bumppaa migraationumeroa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-2288

jarnovayrynen committed:
Korjaa rikkoontunut ylläpitotesti

Kun ylläpitokohde siirrettiin vuodelle 2020, muuttui sinä aikana vaatimukset
siten, että ylläpitokohteella (pääkohde) ei saa olla kaistaa eikä ajorataa.

jarnovayrynen committed:
Korjaa testit

jarnovayrynen committed:
Korjaa vanhan potin e2e-testi

jarnovayrynen committed:
Korjaa testejä

jarnovayrynen committed:
Korjaa takuupvm testeistä

jarnovayrynen committed:
Korjaa takuupvm testeistä

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-2288

jarnovayrynen committed:
Korjaa POT cypress testiä

erno commented on test/clj/harja/palvelin/integraatiot/api/yllapitokohteet_test.clj:
```diff
@@ -734,7 +735,7 @@
                                         payload)]
     (is (not= 200 (:status vastaus)))
-    (is (str/includes? (:body vastaus) "{\"virheet\":[{\"virhe\":{\"koodi\":\"viallisia-tieosia\",\"viesti\":\"-----------\\nMuukohde\\nKohteenosa on päällekkäin osan \\\"Ouluntien kohdeosa\\\" kanssa\\nKohteenosa on päällekkäin toisen osan kanssa\\n\"}}]}"))))
+    (is (str/includes? (:body vastaus) "{\"virheet\":[{\"virhe\":{\"koodi\":\"viallisia-tieosia\",\"viesti\":\"-----------\\nMuukohde\\nKohteenosa on päällekkäin osan \\\"Puolangantien kohdeosa\\\" kanssa\\nKohteenosa on päällekkäin toisen osan kanssa\\n\"}}]}"))))
```
Tässä luotetaan json-mapin avainten järjestykseen, joka ei varmaan ole taattu.

jarnovayrynen replied:
Hyvä kommentti. Käärin tän (dotimes [n 1000 ja ajoin, niin passiin se menee yhä. Periaatteessa olisi voinut odottaa koodi ja viesti -kenttien järjestykseltä satunnaisuutta.

erno commented on src/cljs/harja/ui/grid/perus.cljs:
```diff
@@ -669,6 +674,7 @@
   :voi-poistaa?                         funktio, joka kertoo, voiko rivin poistaa
   :voi-lisata?                          voiko rivin lisätä (boolean)
   :voi-kumota?                          Jos false, ei näytetä kumoa-nappia. Oletus: true.
+  :custom-toiminto                      Muokkauspaneeliin vietävä custom-toiminto
```
Lyhyt tila jos pitää muiden kaltaisena, mutta jos vaan keksii niin tarkempi kuvaus, esim avainten nimet vois olla hyvä.

jarnovayrynen replied:
En ihan saa kiinni tästä - siis custom-toiminto nimen sijaan "lyhyt-tila" tai "avainten-nimet" parametrin nimeksi? Tuolla custom-toiminnolla vois siis passata gridin paneeliin oman toiminnon, tähän tyyliin: 

:custom-toiminto {:teksti "Luo uusi massa"
                      :toiminto #(e! (tiedot-massa/->UusiMassa true))
                      :opts {:ikoni (ikonit/livicon-plus)
                             :luokka "napiton-nappi"}}}

rckopteri commented on dev-resources/less/pages/paallystys_ja_paikkaus.less:
```diff
@@ -20,4 +20,84 @@
+    }
+    .livi-grid {
+        .panel-heading, .panel-body {
+            background-color: #ffffff;
```
Yhtenäisemmän linjan säilyttämiseksi voisi ottaa tämänkin tuolta colors.lessistä: @white

jarnovayrynen replied:
hyvä, laitanpa heti ton @white

rckopteri commented on src/clj/harja/kyselyt/pot2.sql:
```diff
@@ -0,0 +1,6 @@
+SELECT m.id, m.nimi, array_agg(ROW(r.id, r.esiintyma))
+    FROM pot2_massa m
+             LEFT JOIN pot2_massa_runkoaine r ON r.pot2_massa_id = m.id
+    WHERE urakka_id = :urakka-id
```
Voisi ottaa taulun nimen mukaan m.urakka_id tyyppisesti

jarnovayrynen replied:
Kiitos, kävi ilmi että koko koodi oli jo kuollutta, koska specql. Poistin.

rckopteri commented on src/cljs/harja/tiedot/urakka/pot2/massat.cljs:
```diff
@@ -0,0 +1,235 @@
+    (assoc app :massat vastaus))
+
+  HaePot2MassatEpaonnistui
+  (process-event [{vastaus :vastaus} app]
```
Voisi laittaa infon, että haku epäonnistui

rckopteri commented on src/cljs/harja/tiedot/urakka/pot2/massat.cljs:
```diff
@@ -0,0 +1,235 @@
+    (assoc app :materiaalikoodistot vastaus))
+
+  HaeKoodistotEpaonnistui
+  (process-event [{vastaus :vastaus} app]
```
tänne myös ehkä info?

jarnovayrynen replied:
Juu, hyvä

rckopteri commented on src/cljs/harja/tiedot/urakka/pot2/massat.cljs:
```diff
@@ -0,0 +1,235 @@
+
+  PaivitaLomake
+  (process-event [{data :data} app]
+    (let [uusiapp (-> app
```
Ehkä riittäisi, jos palauttaisi sukkana tuon updateni ilman letattua arvoa ja threading macroa

rckopteri commented on src/cljs/harja/tiedot/urakka/pot2/massat.cljs:
```diff
@@ -0,0 +1,235 @@
+  PaivitaAineenTieto
+  (process-event [{polku :polku arvo :arvo} app]
+    (log "PaivitaAineenTieto polku " (pr-str polku), " arvo:" (pr-str arvo))
+    (let [uusiapp (assoc-in app
```
sama kuin yllä

jarnovayrynen committed:
Tee kommenttien pohjalta parannuksia, poista kuollutta koodia

rckopteri commented on src/cljs/harja/tiedot/urakka/pot2/validoinnit.cljs:
```diff
@@ -0,0 +1,120 @@
+
+(defn- pakolliset-runkoaineen-kentat [tyyppi]
+  (case tyyppi
+    (1 2 4 5 6)
```
En ole koskaan tiennyt, että casea voi käyttää näin. Nice!

rckopteri commented on src/cljs/harja/tiedot/urakka/pot2/validoinnit.cljs:
```diff
@@ -0,0 +1,120 @@
+(defn- hae-sideaineen-puutteet [aineet]
+  (into {}
+        (map (fn [[_ aine]]
+               (do
```
Tarvitaanko Do:ta ?

jarnovayrynen replied:
ei tarvi, kiitos taas. Jää noihin looppeihin joskus tuo do, kun tykkään lokitella tuolla sisällä.

rckopteri commented on src/cljs/harja/ui/validointi.cljs:
```diff
@@ -392,3 +393,16 @@
             s))
         skeema))
+
+(defn validoi-numero
```
Käytetäänkö tätä jossakin? Näkisin mieluusti testin tällaiselle validointi funkkarille. 

Haa oli niin iso tiedosto, että se oli kollapsoitu, niin haku ei löytänyt. Mutta ehkä testi?

jarnovayrynen replied:
Joo, testi olis hyvä. katotaanpa joko tähän Pulleroon leivon sen

jarnovayrynen replied:
Lisäsin testin

rckopteri commented on src/cljs/harja/views/urakka/pot2/massalistaus.cljs:
```diff
@@ -0,0 +1,447 @@
+    ;; ja ainekohtaisia eroja käsitellään tässä contains? funktion avulla
+    (remove
+      nil?
+      [(when-not (contains? #{3 7} aineen-koodi)
```
Nerokas ratkaisu. 👍

rckopteri commented on src/cljs/harja/views/urakka/pot2/massalistaus.cljs:
```diff
@@ -0,0 +1,447 @@
+                                                                    (not= (::pot2-domain/koodi %)
+                                                                          (:sideaine/tyyppi sideaine)))
+                                                              sideainetyypit))]
+                      ^{:key (str idx)}
```
Reactin elementin key:hin voisi lisätä indexin lisäksi jonkun sanallisen termin  tyyliin (str "sideaine-" idx) , niin listaustilanteissa samalla sivulla ei vahingossa tule samaa indexiä useamman kerran.

rckopteri commented on src/cljs/harja/views/urakka/pot2/massalistaus.cljs:
```diff
@@ -0,0 +1,447 @@
+   (for [t (ainevalintalaatikot tyyppi aineet)]
+      (let [polun-avaimet [(keyword "harja.domain.pot2" (name tyyppi)) (::pot2-domain/koodi t)]
+            {:keys [valittu?] :as tiedot} (get-in rivi (cons :data polun-avaimet))]
+        ^{:key t}
```
Voisko "t":stä ottaa hashin, niin html koodiin tulisi vähempi tavaraa? tyyliin (hash t)

rckopteri commented on src/cljs/harja/views/urakka/pot2/massalistaus.cljs:
```diff
@@ -0,0 +1,447 @@
+          {:width "300px"}
+          "KM-arvo" kuulamyllyarvo
+          "Litteysluku" litteysluku]]
+        [:span {:style {:color "#004D99"
```
Voisi pohtia, josko tyyliin siirtäisi less tiedostoon jonkun kuvaavan class nimen alle

rckopteri commented on src/cljs/harja/views/urakka/pot2/massalistaus.cljs:
```diff
@@ -0,0 +1,447 @@
+   (for [aine (reverse
+                (sort-by :lisaaine/pitoisuus
+                         (:harja.domain.pot2/lisaaineet rivi)))]
+     ^{:key (:lisaaine/id aine)}
```
Lisäisin :key:hin vielä jotain erottelevaa

rckopteri commented on tietokanta/src/main/resources/db/migration/V1_847__.sql:
```diff
@@ -0,0 +1,335 @@
+-- Vähä inserttejä helpottavaa settia
```
Nämä kommentit kehtaa poistaa

jarnovayrynen replied:
Hyvä! Nää olikin jo monta kertaa iltalenkeillä mielessä, että pitää poistaa ennen mergeä :D

rckopteri commented on tietokanta/src/main/resources/db/migration/V1_847__.sql:
```diff
@@ -0,0 +1,335 @@
+    DoP_nro           TEXT, -- DoP_nroa ei ole aina saatavilla heti? Siksi ei NOT NULL
+
+    -- muokkausmetatiedot
+    poistettu         BOOLEAN   DEFAULT FALSE,
```
Ehkä NOT NULL. lisällä?

rckopteri commented on tietokanta/src/main/resources/db/migration/V1_847__.sql:
```diff
@@ -0,0 +1,335 @@
+(
+    id            SERIAL PRIMARY KEY,
+    pot2_massa_id INTEGER NOT NULL REFERENCES pot2_massa (id),
+    "lopputuote?" BOOLEAN DEFAULT TRUE, -- FALSE = lisätty sideaine
```
Ehkä not null lisällä?

rckopteri commented on tietokanta/testidata/yllapito/paallystysmassat.sql:
```diff
@@ -0,0 +1,36 @@
+INSERT INTO pot2_massa (urakka_id, tyyppi, max_raekoko, kuulamyllyluokka, litteyslukuluokka, DoP_nro, luoja, luotu)
```
Nämä voisi wräpätä 
DO
$$
    DECLARE
   BEGIN

Tyyppiseen funktioon, jolloin voidaan DECLAREN sisään tallentaa urakan id ja sitä ei tarvitsisi hakea monta kertaa.
Tästä olisi ainakin seuraavat hyödyt:
- Helpompi lukea
- Nopeampi ajaa
- Nopeampi ottaa haltuun, jos haluaa muokata

jarnovayrynen committed:
Siisti kommenttien perusteella, poista kuollutta koodia

jarnovayrynen committed:
Korjaa e2e testiä

jarnovayrynen committed:
Poista turhia muotoilumuutoksia

jarnovayrynen committed:
Kommenttien pohjalta parannuksia

jarnovayrynen committed:
Lisää fronttitesti numeron validoinnille

jarnovayrynen committed:
Pakota pari clickiä pot e2e testissä

jarnovayrynen committed:
Kokeile valita vuotta 2017 pot testissä

jarnovayrynen committed:
Palauta piilota kartta e2e-testiin
